developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,134 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+Why UUID is needed
+******************
+
+When developing a new audio signal processing component, we
+traditionally implement the component driver with a new unique component
+type declared together, the painpoints with this method including ABI
+alignment among topology/driver/firmware for each new added components,
+and there could be collision if versions are not aligned.
+
+UUIDs (Universally Unique Identifiers) provide a more scalable and
+collisionn-free way of component identification. UUIDs are currently
+used as the standard interface by all users of the SOF firmware,
+including the tracing subsystem, the topology .m4 files, and the Linux
+topology driver.
+
+Allocating a new UUID for the new added component is mandatory, and
+UUIDs are expected to replace the component type in the near future. For
+the whole SOF stack, the usage of UUID should follow rules as below:
+
+UUID allocation
+***************
+The UUID allocation of a specific component should use the version 4 in
+https://en.wikipedia.org/wiki/Universally_unique_identifier, and the
+value should be specified in the firmware component dirver, e.g. for src
```

Comment:
This second part might be replaced by a reference to the tutorial to maintain a single copy of code samples (easier to maintain).

Reply:
the PR is updated with this part changed, mark it 'resolved'.